src/api/helpers.jl

```diff
@@ -224,6 +224,49 @@ function h5f_get_vfd_handle(file_id, fapl)
     return file_handle[]
 end
 
+"""
+    h5f_get_free_sections(file_id, type, [sect_info::AbstractVector{H5F_sect_info_t}])::AbstractVector{H5F_sect_info_t}
+
+Return an `AbstractVector` of the free section information. If `sect_info` is not provided a new `Vector` will be allocated and returned.
+If `sect_info` is provided, a view, a `SubArray`, will be returned.
+"""
+function h5f_get_free_sections(file_id, type)
+    nsects = h5f_get_free_sections(file_id, type, 0, C_NULL)
+    sect_info = Vector{H5F_sect_info_t}(undef, nsects)
+    if nsects > 0
+        h5f_get_free_sections(file_id, type, nsects, sect_info)
+    end
+    return sect_info
+end
+
+function h5f_get_free_sections(file_id, type, sect_info::AbstractVector{H5F_sect_info_t})
+    nsects = length(sect_info)
+    nsects = h5f_get_free_sections(file_id, type, nsects, sect_info)
+    return @view(sect_info[1:nsects])
+end
```

Comment:
The 3-arg version takes an `AbstractVector` as the third argument and will return a view, a `SubArray`.

Reply:
Ok let me get this straight. If a user passes in a vector of length `n` where `n > nsects` this will fill in those entries just fine. Now if a user passes in a vector with `n< nsects` this will still work as only those `n` entries will be filled.  I think that's right?

Reply:
Correct

Reply:
Ok great, thanks for changing this and the other helper method.

Reply:
Makes sense to me and is closer with the hdf5 library signature which this file aims to simplify for the Julia side. So I think they're both positive changes.